modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -312,31 +314,37 @@ public MenuBarSkin(final MenuBar control) {
                 setFocusedMenuIndex(0);
                 openMenuButton.setHover();
             }
-         };
+        };
+        weakMenuBarFocusedPropertyListener = new WeakChangeListener(menuBarFocusedPropertyListener);
+
         weakSceneKeyEventHandler = new WeakEventHandler<KeyEvent>(keyEventHandler);
-        Utils.executeOnceWhenPropertyIsNonNull(control.sceneProperty(), (Scene scene) -> {
+        executeOnceWhenPropertyIsNonNull(control.sceneProperty(), (Scene scene) -> {
             scene.addEventFilter(KeyEvent.KEY_PRESSED, weakSceneKeyEventHandler);
         });
 
         // When we click else where in the scene - menu selection should be cleared.
         mouseEventHandler = t -> {
-            Bounds containerScreenBounds = container.localToScreen(container.getLayoutBounds());
-            if (containerScreenBounds == null || !containerScreenBounds.contains(t.getScreenX(), t.getScreenY())) {
-                unSelectMenus();
+            if (getSkinnable() != null) {
+                Bounds containerScreenBounds = container.localToScreen(container.getLayoutBounds());
```

Comment:
this null check is wrong: skinnable is guaranteed to never be null - except after dispose, at which time it's illegal to access any of the skin's methods/fields/state. So checking here is smearing over a precondition violation of the caller (most probably a dangling handler/listener) which has to be fixed.

Note we deliberately removed all null checks in all (cleaned) skins :) If we see any exceptions, the procedure is to 

- write a test exposing the exception (it's failing), this test belongs into SkinCleanTest
- find and fix the violator
- see the test passing

There might be other exceptions as well (like IOOB or similar) which require a similar procedure, see the available tests. Actually, I think it's a good idea to actively look out for possible macroscopic misbehavior if handlers/listeners are not removed properly. SkinCleanupTest has examples.

Reply:
agreed.